### PR TITLE
CLOUDSTACK-8773 : NPE in CheckRouterTask, when a DomainRouter happens…

### PR DESCRIPTION
**Pull request by nlivens**
… to be expunged at the same time

### CODE REVIEW

File: engine/schema/src/com/cloud/vm/dao/DomainRouterDaoImpl.java [hunk 1]
```diff
@@ -251,8 +251,11 @@ public boolean remove(final Long id) {
         sc.setJoinParameters("networkRouter", "type", Network.GuestType.Isolated);
         final List<DomainRouterVO> routerIds = listBy(sc);
         final List<DomainRouterVO> routers = new ArrayList<DomainRouterVO>();
-        for (final DomainRouterVO router : routerIds) {
-            routers.add(findById(router.getId()));
+        for (DomainRouterVO router : routerIds) {
+            router = findById(router.getId());
```
Comment (karuturi) [inline engine/schema/src/com/cloud/vm/dao/DomainRouterDaoImpl.java]: Can you use `CollectionUtils.addIgnoreNull(routers,findById(router.getId()))`?

**Comment by karuturi** [comment]
A unittest to verify this case would be nice.

**Comment by bhaisaab** [comment]
LGTM, you can also use the method @karuturi has suggested.

**Comment by nlivens** [comment]
@karuturi I've used the method you suggested, this problem occurs in a specific scenario which is hard to reproduce in a consistent way.

**Comment by karuturi** [comment]
@nlivens thanks for the update. :+1: 
will merge in sometime once travis completes. I assigned you CLOUDSTACK-8773. please resolve it once this is merged.

**Pull request merged**